lib/salesforceiq.js

```diff
@@ -447,6 +448,196 @@ var SalesforceIQ = (function() {
     makeRequest('lists/' + listId, req, cb);
   };
 
+  /**
+   * Format fields for list item to match SalesforceIQ API specs,
+   * using ids pulled from the associated list
+   * (c.f. https://api.salesforceiq.com/#/curl#documentation_lists_list-items_list-item-object)
+   *
+   * @param {string}                  listId  Reference id for the list containing these fields
+   * @param {Object.<string, string>} data    Simple key/value store of field data
+   * @param {Function}                cb      Callback returning error info and formatted fields
+   */
+  SalesforceIQ.prototype.formatListFields = function(listId, data, cb) {
+    this.getList(listId, function(err, list) {
+      if (err) {
+        // REF console.log(err);
+        cb(err);
+      } else {
+        var fields = list.fields;
+        var fieldNameIds = {};
+        var fieldOptionIds = {};
+
+        _.forEach(fields, function(field) {
+          fieldNameIds[field.name] = field.id;
+
+          var optionIds = {};
+          _.forEach(field.listOptions, function(option) {
+            optionIds[option.display] = option.id;
+          });
+
+          if (optionIds) {
+            fieldOptionIds[field.name] = optionIds;
+          }
+        });
+
+
+        var fieldValues = {};
+
+        var invalidFieldNames = [];
+        var invalidFieldOptions = {};
+
+        _.forEach(data, function(value, name) {
+          var nameId = fieldNameIds[name];
+          if (nameId) {
+            var pickList = fieldOptionIds[name];
+            if (!_.isEmpty(pickList)) {
+              var valueId = pickList[value];
+              if (valueId) {
+                value=valueId;
```

Comment:
let's put a space before and after =. It's definitely aesthetic, but is going to be part of our standards in ESLint, so might as well get started!